content/topics/java-specific/oop-basics/_index.md

```diff
@@ -1,15 +1,179 @@
 ---
 title: Java OOP basics resources and readings
+ready: true
 ---
 
-TODO:
+## OOP  in Java
+The main aim of OOP is to bind together the data and the functions that operate on them.This is done so that no other part of the code can access this data except that function.
 
-- class
-- calling functions
-- instantiation of classes
+There are four OOP priciples in java:
```

Comment:
Not really just 4. There are 4 main ones and then there are others.

Reply:
Corrected to four main ones